cfApplication.go

```diff
@@ -40,21 +43,28 @@ type Metadata struct {
 // Routes represents a slice of Routes
 type Routes []Route
 
-// Route captures the key elements that define a Route: fqdn, protocol and port. These values
+// Route captures the key elements that define a Route in a string that maps to a URL structure. These values
 // are captured as runtime routes, meaning that if the CF Application manifest is configured to disable all routes
 // with the `no-route` value, it will translate into an empty slice.
 // By default CloudFoundry will always attempt to create a route for each application, unless specified by the field `no-route` when true
 // For further details check: https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#no-route
 // and https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#random-route
+// Example
+// ---
+//
+//	...
+//	routes:
+//	- route: example.com
+//	  protocol: http2
+//	- route: www.example.com/foo
+//	- route: tcp-example.com:1234
 type Route struct {
-	// FQDN captures the Fully Qualified Domain Name of the hostname field in the route. If the hostname contained a port
+	// URL captures the Fully Qualified Domain Name of the hostname field in the route. If the hostname contained a port
 	// its value it captured in the `Port` field in the Route structure.
-	FQDN string `json:"fqdn"`
-	// Protocol captures the protocol type: http, http2 or tcp.
+	URL string `json:"url"`
+	// Protocol captures the protocol type: http, http2 or tcp. Note that the CF `protocol` field is only available
+	// for CF deployments that use HTTP/2 routing.
 	Protocol RouteProtocol `json:"protocol"`
-	// Port captures the port to use for the route. For RouteProtocol `http`` it is 80; for `http2` it's 443,
-	// and for `tcp` it is as defined in the CF application manifest.
-	Port uint `json:"port"`
```

Comment:
In `URL` field you say `If the hostname contained a port its value it captured in the `Port` field in the Route structure.` but here you remove the `Port` field

Reply:
Reworded comment in URL.